components/consentRequestForm/dateInput/index.jsx

```diff
@@ -0,0 +1,140 @@
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* eslint-disable react/jsx-one-expression-per-line */
+
+import React, { useContext, useEffect, useRef } from "react";
+import { Button, Icons } from "@inrupt/prism-react-components";
+import { format } from "date-fns";
+import T from "prop-types";
+import { DatePicker } from "@material-ui/pickers";
+import { makeStyles } from "@material-ui/styles";
+import { createStyles, IconButton, InputBase } from "@material-ui/core";
+import { useBem } from "@solid/lit-prism-patterns";
+import ConsentRequestContext from "../../../src/contexts/consentRequestContext";
+import styles from "../styles";
+
+export const TESTCAFE_ID_DATE_INPUT = "date-input-field";
+export const TESTCAFE_ID_FOREVER_BUTTON = "forever-button";
+export const TESTCAFE_ID_DATE_PICKER_CALENDAR_BUTTON =
+  "date-picker-calendar-button";
+
+const useStyles = makeStyles((theme) => createStyles(styles(theme)));
+
+export default function DateInput(props) {
+  const {
+    selectedDate,
+    setSelectedDate,
+    datepickerOpen,
+    setDatepickerOpen,
+    handleDateChange,
+    setDateForever,
+  } = props;
+  const classes = useStyles();
+  const bem = useBem(classes);
+  const { consentRequest } = useContext(ConsentRequestContext);
+  const expirationDate = consentRequest?.expirationDate;
+  // FIXME: we will later fetch the expiry date from the consent details
+  const node = useRef();
+
+  const handleClick = (e) => {
+    if (node?.current?.contains(e.target)) {
+      // inside click
+      return;
+    }
+    // outside click
+    setDatepickerOpen(false);
+  };
+
+  useEffect(() => {
+    if (expirationDate) {
+      setSelectedDate(expirationDate);
+    }
+  }, [expirationDate, setSelectedDate]);
+
+  useEffect(() => {
+    // add when mounted
+    document.addEventListener("mousedown", handleClick);
+    // return function to be called when unmounted
+    return () => {
+      document.removeEventListener("mousedown", handleClick);
+    };
+  });
+
+  return (
+    <div ref={node} className={bem("date-container")}>
+      <InputBase
+        classes={{ root: bem("date-input") }}
+        placeholder="Consent expiry date"
+        value={selectedDate ? format(selectedDate, "MMMM' 'd', 'Y") : "Forever"}
```

Comment:
If we're only using the date function here, is it possible for us to format the date manually instead of adding a library just for this?

Reply:
we could of course. The format library is added as part of the dependencies for the `material-ui/pickers` so it wasn;t an extra addition however, it is up for us to decide if we just wanted to format the date manually

Reply:
Ah ok, I didn't know it was a dependency. How annoying.